src/coldfront_plugin_cloud/openshift.py

```
@@ -146,20 +146,79 @@ def create_project(self, suggested_project_name):
+
+    def _get_moc_quota_from_resourcequotas(self, project_id):
+        """This returns a dictionary suitable for merging in with the
+        specification from Adjutant/ColdFront"""
+        resourcequotas = self._openshift_get_resourcequotas(project_id)
+        moc_quota = {}
+        for rq in resourcequotas:
+            name, spec = rq["metadata"]["name"], rq["spec"]
+            logger.info(f"processing resourcequota: {project_id}:{name}")
+            scope_list = spec.get("scopes", [""])
+            for quota_name, quota_value in spec.get("hard", {}).items():
+                for scope_item in scope_list:
+                    moc_quota_name = f"{scope_item}:{quota_name}"
+                    moc_quota.setdefault(moc_quota_name, quota_value)
+        return moc_quota
 
     def get_quota(self, project_id):
-        url = f"{self.auth_url}/projects/{project_id}/quota"
-        r = self.session.get(url)
-        return self.check_response(r)
+        quota_from_project = self._get_moc_quota_from_resourcequotas(project_id)
+
+        quota = {}
+        for quota_name, quota_value in quota_from_project.items():
+            if quota_value:
+                quota[quota_name] = quota_value
+
+        quota_object = {
+            "Version": "0.9",
+            "Kind": "MocQuota",
+            "ProjectName": project_id,
+            "Quota": quota,
+        }
+        return quota_object
+    
+    def _get_moc_quota_used_from_resourcequotas(self, project_id):
+        resourcequotas = self._openshift_get_resourcequotas(project_id)
+        moc_quota_used = {}
+        for rq in resourcequotas:
+            moc_quota_used.update(rq["status"]["used"])
+        return moc_quota_used
+    
+    def get_quota_used(self, project_id):
+        resourcequotas = self._openshift_get_resourcequotas(project_id)
+        moc_quota_used = {}
+        # TODO Any concerns about this being a list? Can a project have multiple resourcequotas?
```

Should the code account for cases where a user's project has more than one ResourceQuota?

---

I just checked and it's possible to create multiple resourcequotas for a namespace but in practice that should be avoided. I'd say if there's more than 1 resourcequota you should just raise an error.

Just posting this for completeness sake. When you have multiple resourcequotas your pods need to meet the limits specified by all
```
➜  ~ oc get resourcequota
NAME                 AGE   REQUEST                        LIMIT
naved-project-rq-1   38m   persistentvolumeclaims: 0/20   limits.cpu: 1/1, limits.memory: 4Gi/8Gi
naved-project-rq-2   38m   persistentvolumeclaims: 0/20   limits.cpu: 1/2, limits.memory: 4Gi/4Gi
```
rq-1 allows 1 CPU and 8 Gi of memory and rq-2 allows 2 CPU and 4 Gi memory, in effect this meant I was restricted to 1 CPU and 4 Gi of memory.